os/include/tinyara/fs/ioctl.h

```diff
@@ -433,6 +434,14 @@
 #define MMINFOIOC_PARSE             _MMINFOIOC(0x0002)
 #define MMINFOIOC_MNG_ALLOCFAIL     _MMINFOIOC(0x0003)
 
+/* Compress driver ioctl definitions ************************/
+#define _COMPIOCVALID(c)    (_IOC_TYPE(c) == _COMPBASE)
+#define _COMPIOC(nr)        _IOC(_COMPBASE, nr)
+
+#define COMPIOC_COMPRESS           _COMPIOC(0x0001)
+#define COMPIOC_GET_COMP_TYPE      _COMPIOC(0x0002)
+#define COMPIOC_DECOMPRESS	   _COMPIOC(0x0003)
```

Comment:
Is it possible to use them by user who does not use compression module?
I can't find any description for them.

Reply:
User need to include sys/ioctl.h

Reply:
Sorry for not enough question.
What I said is
when user wants to use the compression driver, each command (compress, get_comp_type, decompress) should be called with some specific arg and return value.
1. compress and decompress needs to give `struct compress header` for arg, return shows operation result, and get operated data from the arg
2. get_comp_type does not need to give the arg for ioctl, user can get the value from the return value.

User definitely does not know above, when user analyzes the code, then he can know.
But it's bad. That's why I told you "I can't find any description for them".

Reply:
i have updated driver/compress.c file for description

Reply:
For compress and decompress commands, user should give input arguments, input_data, input_data_size. User should allocate memory for input_data. How about output_data? User should allocate? or driver will do?
How about output_size? User should give that for limitation? or driver will give?

I know them because I check the code. But user will not check then you should give exact usages.

Reply:
i have added a README file for that and updated the code for all scenarios